src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -48,38 +57,129 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
     }
   }
 
-  /** Get historical prices for the symbol, range, and date combination
+  /** Get historical prices for the symbol, range, and date combination.
+   * Store values for quicker retrieval.
    * @param symbol a valid symbol
    * @param range an optional range to check
    * @param date an optional date to check; has priority over range
    */
-  public List<IexHistoricalPrices> getHistoricalPricesForSymbols(final String symbol, final String range, final String date) throws Exception{
+  public List<IexHistoricalPrice> getHistoricalPricesForSymbols(
+      final String symbol,
+      final String range,
+      final String date) throws Exception {
 
-    // Throw an error if an invalid symbol was provided
-    if(StringUtils.isBlank(symbol)) {
+    List<IexHistoricalPrice> result = new ArrayList<IexHistoricalPrice>();
+    List<IexHistoricalPrices> hps = new ArrayList<IexHistoricalPrices>();
+
+    // No symbol provided
+    if (StringUtils.isBlank(symbol)) {
       throw new EntityNotFoundException(symbol.getClass(), symbol);
     }
 
-    // Call the endpoint based on input. Date has priority when both date and range are provided.
-    if(StringUtils.isNotBlank(range) && StringUtils.isNotBlank(date)) {
-      return iexClient.getHistoricalPricesForSymbolDateAndRange(symbol, range, date);
-    } else if(range == null && date == null) {
-      return iexClient.getHistoricalPricesForSymbol(symbol);
-    } else if(StringUtils.isNotBlank(date)) {
-       return iexClient.getHistoricalPricesDate(symbol, date);
-    } else if(date == null && range != null) {
-      if(range.equals("")) {
-        throw new HttpMessageNotReadableException("\"range\" is not allowed to be empty");
+    // Retrieve for date if both date and range are provided
+    if (StringUtils.isNotBlank(range) && StringUtils.isNotBlank(date)) {
+
+      hps.addAll(fetchHistoricalPricesDate(symbol, date));
+
+      if (hps.isEmpty()) {
+        result.addAll(iexClient.getHistoricalPricesForSymbolDateAndRange(symbol, range, date));
+        saveIexHistoricalPrices(new IexHistoricalPrices(result.get(0)));
+      } else {
+        result.add(new IexHistoricalPrice(hps.get(0)));
```

Comment:
What happens if there is more than one result for a given date?

Reply:
I resolved this by adding a helper method to get the latest instance